lib/csv/table.rb

```diff
@@ -1028,19 +1028,22 @@ def dig(index_or_header, *index_or_headers)
     end
 
     # :call-seq:
-    #   table.inspect => string
+    #   table.inspect(row_counts) -> self
     #
-    # Returns a <tt>US-ASCII</tt>-encoded \String showing table:
-    # - Class: <tt>CSV::Table</tt>.
-    # - Access mode: <tt>:row</tt>, <tt>:col</tt>, or <tt>:col_or_row</tt>.
-    # - Size:  Row count, including the header row.
-    #
-    # Example:
+    # Return table with specified rows data with header
     #   source = "Name,Value\nfoo,0\nbar,1\nbaz,2\n"
     #   table = CSV.parse(source, headers: true)
-    #   table.inspect # => "#<CSV::Table mode:col_or_row row_count:4>"
-    def inspect
-      "#<#{self.class} mode:#{@mode} row_count:#{to_a.size}>".encode("US-ASCII")
+    #   table.inspect(2) # => #<CSV::Row "Name":"foo" "Value":"0">, #<CSV::Row "Name":"bar" "Value":"1">
+    #
+    def inspect(row_counts)
+      array = [headers.to_csv]
+      if row_counts.nil?
+        return array.join("")
+      end
+      @table.first(row_counts).each do |row|
+        array.push(row.fields.to_csv) unless row.header_row?
+      end
```

Comment:
How about adding `:limit` option to `CSV::Table#to_csv` and use it from `#inspect?`

```ruby
    def to_csv(write_headers: true, limit: nil, **options)
      array = write_headers ? [headers.to_csv(**options)] : []
      limit ||= @table.size
      limit = @table.size + 1 - limit if limit < 0 
      @table.first(limit) do |row|
        array.push(row.fields.to_csv(**options)) unless row.header_row?
      end

      array.join("")
    end

    def inspect
      inspected = +"#<#{self.class} mode:#{@mode} row_count:#{to_a.size}>"
      summary = to_s(limit: 5)
      inspected << "\n" << summary if summary.encoding.ascii_compatible?
      inspected
    end
```